src/pyird/image/wavcal.py

```diff
@@ -1,11 +1,85 @@
+from pyird.io.read_linelist import read_linelist
 from astropy.io import fits
 import numpy as np
 from scipy.signal import medfilt
+from scipy.optimize import leastsq
 import pandas as pd
 
+import matplotlib.pyplot as plt
+
+def pdat2matrix(pdat):
+    data = np.zeros((l-j,2048))
```

Comment:
can you use a parameter for 2048 to be able to change the detector?
like
```python
pdat2matrix(pdat, npix=2048)
    data = np.zeros((l-j,npix))
```

Reply:
I adopted npix. In addition, I modified the shape of matrix (order, pixel) -> (pixel, order).